### PR TITLE
 MAINT/DOC: Show the location of an empty list in np.block

### PR DESCRIPTION
**Pull request by eric-wieser**
Spun out of #11971

### CODE REVIEW

**Comment by mattip** [comment]
Looks OK to me - refactoring + doc change + extra check that seems to be hit in a test

**Review by mattip — APPROVED** [review]

File: numpy/core/shape_base.py [hunk 1]
```diff
@@ -361,6 +361,11 @@ def stack(arrays, axis=0, out=None):
     return _nx.concatenate(expanded_arrays, axis=axis, out=out)
 
 
+def _block_format_index(index):
+    idx_str = ''.join('[{}]'.format(i) for i in index if i is not None)
```
Comment (charris) [inline numpy/core/shape_base.py]: Needs docstring. Would be good to know what the function is intended to do.

**Comment by eric-wieser** [comment]
That helper function existed before and just moved, but docstring added anyway

**Comment by charris** [comment]
Great, thanks. There are a number of helper functions floating about the need docstrings, it does help long term maintenance.

**Review by charris — APPROVED** [review]

**Comment by charris** [comment]
Skipping Shippable. Thanks Eric.

**Pull request merged**